src/pk_core.c

```diff
@@ -648,7 +648,37 @@ DEF(coreExit,
 // String functions.
 // -----------------
 
-// TODO: substring.
+DEF(coreStrSub,
+  "str_sub(str:string, pos:num, len:num) -> string\n"
+  "Returns a substring from a given string supplied. In addition, "
+  "the position and length of the substring are provided when this "
+  "function is called. For example: `str_sub(str, pos, len)`.") {
+  int c = 0;
+  int64_t len, pos;
+
+  String* str = NULL;
+  if (!validateArgString(vm, 1, &str)) return;
+  if (!validateInteger(vm, ARG(2), &pos, "Argument 2")) return;
+  if (!validateInteger(vm, ARG(3), &len, "Argument 3")) return;
+
+  if (str->length < 1)
+    RET_ERR(newString(vm, "Expected a string length more than 0."));
```

Comment:
We should allow an empty string to be used to get substring too. otherwise, the user has to check if the string is empty or not every time they calling the substring function.

Also we allow the `length` to be `0` for the same reason (example below)

```c
  String* str = NULL;
  if (!validateArgString(vm, 1, &str)) return;
  if (!validateInteger(vm, ARG(2), &pos, "Argument 2")) return;
  if (!validateInteger(vm, ARG(3), &len, "Argument 3")) return;

  // End case.
  if (len == 0) RET(newStringLength(vm, "", 0));
```